lib/matplotlib/cbook/__init__.py

```diff
@@ -1045,28 +1028,14 @@ def __iter__(self):
         The iterator is invalid if interleaved with calls to join().
         """
         self.clean()
-        token = object()
-
-        # Mark each group as we come across if by appending a token,
-        # and don't yield it twice
-        for group in self._mapping.values():
-            if group[-1] is not token:
-                yield [x() for x in group]
-                group.append(token)
-
-        # Cleanup the tokens
-        for group in self._mapping.values():
-            if group[-1] is token:
-                del group[-1]
+        for group in ({id(group): group for group in self._mapping.values()}
+                      .values()):
+            yield [x() for x in group]
 
     def get_siblings(self, a):
-        """
-        Returns all of the items joined with *a*, including itself.
-        """
+        """Returns all of the items joined with *a*, including itself."""
         self.clean()
-
-        siblings = self._mapping.get(ref(a), [ref(a)])
-        return [x() for x in siblings]
+        return [x() for x in self._mapping.get(ref(a), [ref(a)])]
```

Comment:
I think it helps if code is a bit self-documenting.  I prefer the way it was before.

Reply:
I agree; I like concise code, but shorter is not always better, and this is a great example of that.  That intermediate variable works wonders for readability.

Reply:
restored previous version